plugins/src/main/java/org/smartregister/fhir/gateway/plugins/SyncAccessDecision.java

```diff
@@ -247,9 +249,41 @@ public String postProcess(RequestDetailsReader request, HttpResponse response)
             resultContent = this.fhirR4JsonParser.encodeResourceToString(practitionerDetailsBundle);
         }
 
+        String resourceType = request.getResourceName();
+
+        if (Constants.SyncStrategy.LOCATION.equals(resourceType)) {
+            String method = request.getRequestType().name();
+            if ("POST".equals(method) || "PUT".equals(method)) {
```

Comment:
I think we should merge these two conditions into one:

```suggestion
        if (Constants.SyncStrategy.LOCATION.equals(resourceType) && ("POST".equals(request.getRequestType().name()) || "PUT".equals(request.getRequestType().name()))) { 
```